packages/sanity/src/core/preview/observeFields.ts

```diff
@@ -104,7 +104,7 @@ export function createObserveFields(options: {
             mergeMap((result) => {
               return concat(
                 observableOf(result),
-                result === undefined // hack: if we get undefined as result here it can be because the document has
+                result === null // hack: if we get null as result here it can be because the document has
```

Comment:
The `null` is coming from here (which never returns `undefined`): https://github.com/sanity-io/sanity/blob/b14622ed44899ad6da7bb0fd1d46f56c0186a04b/packages/sanity/src/core/preview/utils/optimizeQuery.ts#L61